prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -80,8 +87,22 @@ def build_agent(
         engine.register(f)
 
     agent.max_tries = 3
+
+    # We restore the prompt from a historical session.
+    if prompt_handler:
+        historical_prompt = prompt_handler.fetch_latest_prompt()
+        system_prompt = historical_prompt.prompt
```

Comment:
This is responsible for overwriting `system_prompt` with the latest value from the DB.